### PR TITLE
Update Contributing.md - clang-format plugin for VS2017

### PR DESCRIPTION
**Pull request by ZexaronS**
Some of the things that I had to figure out on my own that I think should be noted, should avoid some of the frustration I had.

### CODE REVIEW

File: Contributing.md [hunk 1]
```diff
@@ -30,10 +30,24 @@ This project uses clang-format (stable branch) to check for common style issues.
...
+  - This operation may produce `.TMP` backups of the original files that were modified. Make sure you do not include these files when restagining. You may move or delete them. To ignore them by git, copy this `*.TMP` into a new line in the `exclude` file in the `.git\info` directory inside your local repository. Note that `.git` is a hidden folder under Windows.
+  - Windows users: To delete all `.TMP` files in your repository, run command prompt from the root repository location with the following command `del /s /q *.tmp`
+  - Windows users: Unfortunately clang-format is only available for official download packed inside a rather large LLVM installer, whereas clang-format is very small. Thankfully there is a way to skip the installer. The LLVM installer executable can be opened by [7-Zip](http://www.7-zip.org/download.html) directly as if it's a ZIP file. Download the LLVM installer but don't run it. Rename the LLVM installer's `.exe` extension to `.7z` or drag&drop it onto `7zFM.exe` located in the 7-Zip root installation directory. Open the `bin` folder, locate `clang-format.exe` and place it in `usr\bin` subdirectory of your Git installation path.
```
Comment (Helios747) [inline Contributing.md]: Remove this. It's not in the scope of this file to tell users how to do stupid things to the LLVM installer.

The above can probably be removed as well. Run correctly, clang-format won't leave .tmp files behind.

Comment (ZexaronS) [reply Contributing.md]: Provide the "correctly" command then.

Comment (Helios747) [reply Contributing.md]: It's currently in the contributing.md file. You're the only person having this problem.

**Comment by stenzek** [comment]
I don't see the point of this. None of the contributing documentation is platform-specific, why introduce it? Besides, if you're a new contributor and haven't set the formatter in your environment, the lint bot will give you a patch that can be applied anyway.

File: Contributing.md (hunk 1 above)
Comment (ZexaronS) [reply Contributing.md]: You assume I'm the only one, for all the future newcomers I guess, and you can't tell me what is the correct way. So as long as you know and you have no problems, this doesn't need to be updated.

I didn't thought these files are meant for the people who are already contributing..

**Comment by Ebola16** [comment]
For what it's worth, I found mentioning the visual studio extension helpful as I just figured out how to use clang-formatting with it. I'd also mention that the clang-formatting extension adds options to the tools menu in Visual Studio.

If anything is added to contributing.md by this PR, it should be that installing LLVM or the Visual Studio extension is required to use clang-formatting on Windows.

**Comment by ZexaronS** [comment]
@Stenzek Is there a rule anywhere these files must not be? I didn't see it. Shouldn't time and frustration saving information override common practices sometimes.

Besides the whole VS2017 part is platform specific by default.  In order to follow your rule I wouldn't be able to even mention the existance of it.

Lastly, why do you guys assume that a newcomer knows git/github at all, I didn't, I felt like on another planet, and still a newcomer in many ways and everyone saw what kind of a crash-landing was with me getting into this. I feel for the other newcomers, bits like this may seem insignificant but shouldn't be underestimated.

**Comment by Helios747** [comment]
Mentioning visual studio's extension to simplify the process is fine.

Everything else about this PR needs to be removed.

**Comment by ZexaronS** [comment]
Oh yes ofcourse, I'll be so thrilled throwing away 2 hours of work. But I'm more upset over the thing being called stupid.

Fortunately, the VSIX method does indeed make the first method unappealing for most if not all Windows users. Let's just hope the thing doesn't stop working, but then you'd wish for a good backup.

**Comment by ZexaronS** [comment]
Thank you, nice attitude.

**Comment by JosJuice** [comment]
Sorry, but GitHub doesn't allow reopening this PR because you force pushed while it was closed. I don't know why GitHub works like that, but it does.

**Comment by ZexaronS** [comment]
Thank you for letting me know.

**Pull request closed without merging**